### PR TITLE
Dev/ft/ring 28301 repository layout

### PR DESCRIPTION
**Pull request by maximelubin**
Setup basic repository structure and unit tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,67 @@
...
+can be found in [hyperdrive project README](https://bitbucket.org/scality/ring/src/f0a3b504523bc0049f8892b1530f3209d0436a1c/?at=feature%2FRING-21232-hyperdrive).
+
+Hyperdrive client functional requirements and specs are inside the [main Zenko architecture document](https://docs.google.com/document/d/1XdOn7MKuWLGb6hd9Apa3Q54uvO5K7InURnskvYH8WVw).
+
+## Usage
+
+### Installation
+
+```shell
+npm install --save scality/sproxydclient
```
Comment (ghost) [inline README.md]: sproxydclient ? :o

File: README.md [hunk 2]
```diff
@@ -1,2 +1,67 @@
...
+```
+
+### Running tests
+To use the linter and run the tests:
+
+```shell
+# ESLint with Scality's custom rules
+npm run-script lint
```
Comment (ghost) [inline README.md]: `npm run lint` is usually enough :)

File: lib/hdclient.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+     * Destroy connections kept alive by the client
+     *
+     * @return {undefined}
+     */
+    destroy() {
+        this.httpAgent.destroy();
+    }
+
+    /**
+     * Create a dedicated logger for Sproxyd, from the provided werelogs API
```
Comment (ghost) [inline lib/hdclient.js]: :D Copy/paset I guess, you're not sproxyd anymore !

File: package.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  "description": "Hyperdrive appliance client library",
+  "main": "index.js",
+  "scripts": {
+    "lint": "eslint $(git ls-files '*.js')",
+    "test": "mocha --use_strict --check-leaks --no-exit "
+  },
+  "repository": "scality/hdclient",
+  "author": "Maxime Lubin",
+  "dependencies": {
+    "werelogs": "scality/werelogs"
```
Comment (ghost) [inline package.json]: Was a model chosen for the dependency handling of hdclient ? Are you going to depend on a fixed version of Werelogs, or are you going to do the same (mistake?) as what was done in our S3C components, depending on moving branches ?

File: package.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  },
+  "devDependencies": {
+    "eslint": "2.13.1",
+    "eslint-config-airbnb": "6.2.0",
+    "eslint-config-scality": "scality/Guidelines",
```
Comment (ghost) [inline package.json]: same remark relatively to scality/* repos

**Review by ghost — COMMENTED** [review]
some sproxyd references are left around, and the dependency handling might deserve better defintion asap. otherwise, lGTM

File: README.md (hunk 1 above)
Comment (maximelubin) [reply README.md]: Damn ><

File: package.json (hunk 4 above)
Comment (maximelubin) [reply package.json]: Frankly I have no idea. I adapted the package.json from Sproxyd... But yeah, seems like a good idea to fix the version.

**Comment by maximelubin** [comment]
@MikaYuoadas if you can find a couple minutes t check eve/ folder. The first build to have done everything I wanted correctly. But idk ieverything was cleaned up properly

File: eve/main.yml [hunk 6]
```diff
@@ -0,0 +1,33 @@
+---
+version: 0.2
+
+branches:
+  user/*, dev/*:
```
Comment (MikaYuoadas) [inline eve/main.yml:5]: Is it on purpose you only build dev/* and user/* branches for pre-merge?
With this yml eve will *not* build the feature/*, bugfix/* branches of the users.
Also Wall-E's q/* and w/* branches won't be automatically built either.

**Review by MikaYuoadas — COMMENTED** [review]
LGTM.
Just a remark on the choice of branch pattern to build if you want to use Wall-E to do the merges at some point. :)

File: eve/main.yml (hunk 6 above)
Comment (maximelubin) [reply eve/main.yml:5]: Ah! feature/, bugfix/ & co are omitted on purpose. As for q/ and w/* I really did forget that. Will add them up

**Pull request merged**